batched/dense/impl/KokkosBatched_Gbtrs_Serial_Impl.hpp

```diff
@@ -0,0 +1,108 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSBATCHED_GBTRS_SERIAL_IMPL_HPP_
+#define KOKKOSBATCHED_GBTRS_SERIAL_IMPL_HPP_
+
+#include <KokkosBatched_Util.hpp>
+#include "KokkosBatched_Gbtrs_Serial_Internal.hpp"
+
+namespace KokkosBatched {
+namespace Impl {
+template <typename AViewType, typename PivViewType, typename BViewType>
+KOKKOS_INLINE_FUNCTION static int checkGbtrsInput([[maybe_unused]] const AViewType &A,
+                                                  [[maybe_unused]] const PivViewType &ipiv,
+                                                  [[maybe_unused]] const BViewType &b, [[maybe_unused]] const int kl,
+                                                  [[maybe_unused]] const int ku) {
+  static_assert(Kokkos::is_view_v<AViewType>, "KokkosBatched::gbtrs: AViewType is not a Kokkos::View.");
+  static_assert(Kokkos::is_view_v<PivViewType>, "KokkosBatched::gbtrs: PivViewType is not a Kokkos::View.");
+  static_assert(Kokkos::is_view_v<BViewType>, "KokkosBatched::gbtrs: BViewType is not a Kokkos::View.");
+  static_assert(AViewType::rank == 2, "KokkosBatched::gbtrs: AViewType must have rank 2.");
```

Comment:
In general, switch to `View::rank()` instead of `View::rank` as explained in the documentation of [Kokkos::View](https://kokkos.org/kokkos-core-wiki/API/core/view/view.html#enums).

Reply:
Sure